spec/bigcommerce/prometheus/servers/thin/server_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe Bigcommerce::Prometheus::Servers::Thin::Server do
+  before do
+    Bigcommerce::Prometheus.reset
+  end
+
+  context 'when the thread pool size is not configured' do
+    it 'should fall back to the default configuration' do
+      server = described_class.new
+      expect(server.threadpool_size).to eq ::Bigcommerce::Prometheus.server_thread_pool_size
+      expect(server.threadpool_size).to eq 20
+    end
+  end
+
+  context 'when we want to customize the thread pool size' do
+    it 'allows you to set the thread pool size through the configuration block' do
+      Bigcommerce::Prometheus.configure do |c|
+        c.server_thread_pool_size = 12
+      end
```

Comment:
💁‍♂️  This belongs in a `before` block. You can also set `12` to a let var to make it easier to compare to below

Reply:
Updated!